proposals/15-message-sending-ux-in-lotus.md

```diff
@@ -0,0 +1,135 @@
+- Previous messages are blocking new messages due to changing network conditions or insufficient funds.
+	- **Solution part 1**:  Warn users when new messages are submitted and ask for confirmation that user wants to proceed.
+	- **Solution part 2**: Build an interactive tool for observing state of messages, allowing the user to reprice them, discard them and showing the user what is the exact issue with them (head of the line blocking, insufficient funds, too low fee)
+
+#### Impact
+<!--\_How directly important is the outcome to web3 dev stack product-market fit?\_-->
+
+🔥🔥 
+This project has potential to decrease attrition of new users that are starting to experiment with Filecoin, while at the same time improving the experience of existing users. 
+
+<!--
+Explain why you have chosen this rating
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+#### Leverage
+\_How much would nailing this project improve our knowledge and ability to execute future projects?\_
+
+🎯
+
+This project develops features needed by other functionalities.
+
+
+<!-- Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+#### Confidence
+\_How sure are we that this impact would be realized? Label from \[this scale\](https://medium.com/@nimay/inside-pro1duct-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)\_.
+<!--Explain why this rating-->
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+- Assemble a prioritized list of message sending issues users have encountered from Lotus issue tracker, user reports, Slack. 
+- Fix issues that can be resolved with detect and warn/fail approach.
+- Design and implement interactive message sending UI. 
+- Design and implement interactive message watching UI - it should fulfill following roles for the user.
+	- Clear indication of whether the message will land on chain in short time
+	- Clear indication of past messages, number of confirmations, exit status.
+	- Provide a way to resolve "struck" messages, interactive functionality of `mpool replace`.
+
+#### What does done look like?
+<!--\_What specific deliverables should completed to consider this project done?\_-->
+Issues identified in the first step are resolved, users are protected against issues mentioned in this document.
+
+####  What does success look like?
+<!--\_Success means impact. How will we know we did the right thing?\_-->
+Number of issues and complaints by users and end users regarding message related interactions with Filecoin is reduced. Message sending related problems are no longer mentioned when on-boarding new users.
```

Comment:
Could we look at metrics like total messages on chain, # clients that have sent at least one message, etc?

Reply:
Total messages won't tell us anything as 99.9% of them are from automated systems.
We can maybe look at: "new first time message senders".